catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -524,12 +539,126 @@ public void init() {
           subscriptions.removeSubscriptions(id);
           return ImmutableMap.of("message", "Successfully deleted.");
         },
         util::getJson);
 
+    get(
+        "/workspace/:id/queries",
+        (req, res) -> {
+          String id = req.params(":id");
+          final Metacard workspace = getWorkspaceMetacard(id);
+
+          Attribute queryAttr =
+              workspace.getAttribute(WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES);
+
+          List<Map<String, Object>> queries = new ArrayList<>();
+
+          if (queryAttr != null) {
+            List<String> ids =
+                queryAttr.getValues().stream().map(String.class::cast).collect(Collectors.toList());
+
+            util.getMetacards(ids, QueryMetacardTypeImpl.QUERY_TAG)
+                .values()
+                .stream()
+                .map(Result::getMetacard)
+                .map(queryTransformer::transformMetacardIntoMap)
+                .forEach(queries::add);
+          }
+
+          return queries;
+        },
+        util::getJson);
+
+    post(
+        "/query",
+        APPLICATION_JSON,
+        (req, res) -> {
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
+
+          Metacard metacard =
+              queryTransformer.transformJsonIntoMetacard((Map<String, Object>) body.get("query"));
+          Metacard saved = saveMetacard(metacard);
+
+          String workspaceId = (String) body.get("id");
+          Metacard workspace = util.getMetacard(workspaceId);
+
+          List<String> queryIds =
+              getWorkspaceAssociations(workspace, WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES);
+          queryIds.add(saved.getId());
+
+          setWorkspaceAssociation(
+              workspace, WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES, queryIds);
+          updateMetacard(workspaceId, workspace);
+
+          return queryTransformer.transformMetacardIntoMap(saved);
+        },
+        util::getJson);
+
+    put(
+        "/query/:id",
+        APPLICATION_JSON,
+        (req, res) -> {
+          String id = req.params("id");
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
+
+          Metacard metacard = queryTransformer.transformJsonIntoMetacard(body);
+          Metacard updated = updateMetacard(id, metacard);
+
+          return queryTransformer.transformMetacardIntoMap(updated);
+        },
+        util::getJson);
+
+    get(
+        "/query/:id",
+        (req, res) -> {
+          String id = req.params("id");
+          Metacard metacard = util.getMetacard(id);
+
+          if (metacard.getTags().contains(QueryMetacardTypeImpl.QUERY_TAG)) {
+            return queryTransformer.transformMetacardIntoMap(metacard);
+          } else {
+            res.status(400);
+            return ImmutableMap.of("message", "requested ID is not a query");
+          }
+        },
+        util::getJson);
+
+    delete(
+        "/query/:workspaceId/:queryId",
```

Comment:
We need to fetch the workspace so we can remove the deleted query from it, and we need the workspace ID to do that.

Reply:
So recent changes have removed the need to get the workspace. There is now no reference of query or workspace IDs on either metacard. They are now stored in an `AssociationMetacard`

Reply:
Long story short the endpoint is back to `/query/:id`. This endpoint makes the assumption that a query only belongs to a single workspace which is true for now. It will require changes in the future if the same query metacard can belong to multiple workspaces